deno/lib/types.ts

```diff
@@ -86,7 +87,7 @@ export abstract class ZodType<
 
     const THROW_ERROR_IF_PRESENT = (key: string) => (data: any) => {
       key;
-      if (!ERROR.isEmpty) throw ERROR;
+      if (ISSUES.length > 0) throw new ZodError(ISSUES);
```

Comment:
@colinhacks I wish this function didn't signal failure by throwing since there's nothing in principle exceptional about validation failing... There could be performance impact in either direction.

I'm not sure if you agree that this is worth investigating, if so this could be a good area for future work (after this PR).